services/sidecar/rollup/derivation/batch_builder.go

```diff
@@ -81,52 +82,66 @@ func (b *batchBuilder) Build() (*BatchAttributes, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to build batch: %w", err)
 	}
-	b.builtBatchAttrs = batchAttrs
-	return batchAttrs, nil
+	batchData, err := encodeBatch(batchAttrs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode batch: %w", err)
+	}
+
+	b.builtBatchData = batchData
+	return b.builtBatchData, nil
 }
 
 func (b *batchBuilder) Advance() {
-	b.builtBatchAttrs = nil
+	b.builtBatchData = nil
 }
 
 func (b *batchBuilder) serializeToAttrs() (*BatchAttributes, error) {
+	if len(b.pendingBlocks) == 0 {
+		return nil, errors.New("no pending blocks")
```

Comment:
we shouldn't return an error here? this will terminate the entire building process i guess?

Reply:
i don't think returning an empty batch here makes sense since there's not value to populate `FirstL2BlockNumber`